### PR TITLE
Rename eventLegal to modernFatefulEncounter

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
Should've been done a long time ago considering it's used essentially the same way in Gen 4.

## **Discord contact info**
AsparagusEduardo#6051

### CODE REVIEW

**Comment by AsparagusEduardo** [comment]
Updated name to `modernFatefulEncounter` ass discussed in the pret Discord server.

File: asm/macros/event.inc [hunk 1]
```diff
@@ -1932,10 +1932,10 @@
...
-	@ Creates an "event legal" Pokémon for an encounter
-	.macro seteventmon species:req, level:req, item=ITEM_NONE
+	@ Creates a Pokémon with the modernFatefulEncounter bit set for an encounter
+	.macro setmodernfatefulencountermon species:req, level:req, item=ITEM_NONE
 	setvar VAR_0x8004, \species
 	setvar VAR_0x8005, \level
 	setvar VAR_0x8006, \item
-	special CreateEventLegalEnemyMon
+	special CreateModernFatefulEncounterEnemyMon
 	.endm
```
Comment (GriffinRichards) [inline asm/macros/event.inc:1941]: These should probably be `seteventmon` and `CreateEnemyEventMon`, it's more descriptive about what they're used for (it's just that event pokemon have the "modern fateful encounter" bit set).

File: data/script_cmd_table.inc [hunk 2]
```diff
@@ -205,8 +205,8 @@ gScriptCmdTable::
 	.4byte ScrCmd_nop1                      @ 0xca
 	.4byte ScrCmd_nop1                      @ 0xcb
 	.4byte ScrCmd_nop1                      @ 0xcc
-	.4byte ScrCmd_setmoneventlegal          @ 0xcd
-	.4byte ScrCmd_checkmoneventlegal        @ 0xce
+	.4byte ScrCmd_setmonmodernfatefulencounter          @ 0xcd
+	.4byte ScrCmd_checkmonmodernfatefulencounter        @ 0xce
 	.4byte ScrCmd_trywondercardscript       @ 0xcf
```
Comment (GriffinRichards) [inline data/script_cmd_table.inc:210]: Alignment

File: src/pokemon.c [hunk 3]
```diff
@@ -4341,8 +4341,8 @@ void SetBoxMonData(struct BoxPokemon *boxMon, s32 field, const void *dataArg)
     case MON_DATA_UNUSED_RIBBONS:
         SET8(substruct3->unusedRibbons);
         break;
-    case MON_DATA_EVENT_LEGAL:
-        SET8(substruct3->eventLegal);
+    case MON_DATA_FATEFUL_ENCOUNTER:
+        SET8(substruct3->modernFatefulEncounter);
```
Comment (GriffinRichards) [inline src/pokemon.c:4345]: `MON_DATA` constants should match the name of their field.

**Comment by AsparagusEduardo** [comment]
Ready for re-review.

**Pull request merged**